### DISCUSSION TITLE
run_method() and passing an object as a parameter

### DISCUSSION THREAD

**Discussion by FearThePenguin** [zauberzeug/nicegui#606] [Q&A] [answered]
I am trying to call the "setPagination" method of a table using run_method but I am unsure how to properly format an "object" as a parameter.

here is a screenshot and link of the documentation I am using.  https://quasar.dev/vue-components/table
![image](https://user-images.githubusercontent.com/4465321/227051556-5a434c90-9e65-4933-ad6c-44c90ac5550f.png)

I have tried 
`table.run_method('setPagination', '{\'sortBy\': \'score\', \'descending\': True, \'rowsPerPage\': 20}')`
and 
`table.run_method('setPagination', {'sortBy': 'score', 'descending': True, 'rowsPerPage': 20})`
with no luck.

Overall my goal is to simply set the default sort order for the table so I am open to alternate solution.

**Comment by falkoschindler** [comment, marked as answer]
That's a good question, @FearThePenguin! Setting a default sort order for table rows should be possible.

It looks like you are trying to call `run_method` right after creating the table. This is problematic, since the client is not yet connected, so there's no one to receive the `run_method` call. Calling it later, e.g. in an event handler, setting the pagination works:

```py
table = ui.table(columns=[
    {'name': 'name', 'label': 'Name', 'field': 'name', 'required': True, 'align': 'left'},
    {'name': 'age', 'label': 'Age', 'field': 'age', 'sortable': True},
], rows=[
    {'name': 'Alice', 'age': 18},
    {'name': 'Bob', 'age': 21},
    {'name': 'Carol', 'age': 20},
], row_key='name')

def sort_by_age():
    table.run_method('setPagination', {'sortBy': 'age', 'descending': True, 'rowsPerPage': 20})
ui.button('Sort by age', on_click=sort_by_age)
```

But if you want to sort the rows right in the beginning, maybe you can simply sort them in Python like this:

```py
columns = [
    {'name': 'name', 'label': 'Name', 'field': 'name', 'required': True, 'align': 'left'},
    {'name': 'age', 'label': 'Age', 'field': 'age', 'sortable': True},
]
rows = sorted([
    {'name': 'Alice', 'age': 18},
    {'name': 'Bob', 'age': 21},
    {'name': 'Carol', 'age': 20},
], key=lambda row: row['age'], reverse=True)
table = ui.table(columns=columns, rows=rows, row_key='name')
```
